x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,199 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+
+	// 5. let epoch 1 pass
+	// ******************** EPOCH 1 *********************
+	// let 1 epoch pass
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// clPoolExternalGaugeIdPool1 expects full because the numEpochPaidOver is 1 for that gagueId
+	//  clPoolExternalGaugeIdPool2 expects half because the numEpochPaidOver is 2 for that gagueId
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 1, halfOfExternalGaugeCoins)
+
+	internalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	// check internalIncentiveCLPool
+	s.Require().Equal(2, len(internalIncentiveClPool))
+	emissionRateForExternalTokens := sdk.NewDecFromInt(sdk.NewInt(defaultExternalGaugeValue)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[0], emissionRateForExternalTokens, internalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[1], emissionRateForExternalTokens, internalIncentiveClPool[1])
+
+	// check externalIncentiveCLPool
+	s.Require().Equal(2, len(externalIncentiveClPool))
+	emissionRateForExternalTokens = sdk.NewDecFromInt(sdk.NewInt(defaultExternalGaugeValue)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * 2).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[0], emissionRateForExternalTokens, externalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[1], emissionRateForExternalTokens, externalIncentiveClPool[1])
+
+	// 6. Remove distribution records for internal incentives using HandleReplacePoolIncentivesProposal
+	s.IncentivizeInternalGauge([]uint64{clPoolId1.GetId(), clPoolId2.GetId()}, epochInfo.Duration, true)
+
+	// 7. let epoch 2 pass
+	// ******************** EPOCH 2 *********************
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.Identifier, 2)
+
+	s.ValidateDistributedGauge(clPoolInternalGaugeId, 2, internalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 2, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 2, externalGaugeCoins)
+
+	internalIncentiveClPoolEpoch2, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPoolEpoch2, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
```

Comment:
Here, it is also unclear to me why one is named internal and the other one external when either pool is both externally and internally incentivized

Reply:
yea i figured this was badly named, i changed it

Reply:
> yea i figured this was badly named, i changed it to this (https://github.com/osmosis-labs/osmosis/pull/5500#discussion_r1235611160)